### PR TITLE
Export default with class name

### PR DESCRIPTION
**Pull request by colevoss**
In the style guide, it says not to use displayName, which I am fine with, but it seems like it discourages the use of `export default` when you can still name a class with
`export default class SomeComponent extends Component {}`.

I think this syntax is better than 
```javascript
class Something {}
export default Something
```

### CODE REVIEW

File: linters/.eslintrc [hunk 1]
```diff
@@ -188,31 +189,6 @@
...
-        "componentWillUpdate",
-        "componentDidUpdate",
-        "componentWillUnmount",
-        "/^on.+$/",
-        "/^get.+$/",
-        "/^render.+$/",
-        "render"
-      ]
-    }]
+    "react/sort-comp": 2,         // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md
```
Comment (ljharb) [inline linters/.eslintrc]: this change seems unrelated to the PR; please revert

File: linters/.eslintrc [hunk 2]
```diff
@@ -144,7 +144,8 @@
         "afterColon": true
     }],
     "new-cap": [2, {                 // http://eslint.org/docs/rules/new-cap
-      "newIsCap": true
+        "newIsCap": true,
+        "capIsNew": false
```
Comment (ljharb) [inline linters/.eslintrc]: this change seems unrelated to the PR; please revert

File: README.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
-[![Gitter](https://badges.gitter.im/Join Chat.svg)](https://gitter.im/airbnb/javascript?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
-
-# Airbnb JavaScript Style Guide() {
+# Hudl JavaScript Style Guide() {
```
Comment (ljharb) [inline README.md]: this change seems unrelated to the PR; please revert

File: react/README.md [hunk 4]
```diff
@@ -89,6 +89,11 @@
     }
  
     export default ReservationCard;
+
+
+    // better
+    export default class ReservationCard extends React.Component {
```
Comment (ljharb) [inline react/README.md]: this is totally great! The only time we don't `export` a `class` inline, default or otherwise, is when we have static properties added prior to export (pending the advancement of the static class properties proposal)

File: README.md (hunk 3 above)
Comment (colevoss) [inline README.md]: Yes, sorry, my company forked this and somehow my first PR attempt went to this repo, so I just kept it up. Didn't see this. I'll change it.

**Comment by colevoss** [comment]
This PR can be deleted/closed. I mean to put this PR on our fork of this repo. I'll submit the same PR to just this repo. SORRY GUYS
